python/ray/data/preprocessor.py

```diff
@@ -1,14 +1,22 @@
 import abc
 import warnings
 from enum import Enum
-from typing import Optional, TYPE_CHECKING
+from typing import Dict, Optional, Union, TYPE_CHECKING
 from ray.util.annotations import PublicAPI
 
 if TYPE_CHECKING:
+    import numpy as np
     import pandas as pd
+    import pyarrow
 
 from ray.data import Dataset
-from ray.air.predictor import DataBatchType
+
+# Duplicated from ray.air.predictor
+# Causes a circular import otherwise
+# TODO move this to one place
+DataBatchType = Union[
```

Comment:
Can you split this into a separate ray.air.data_batch_type file like here https://github.com/ray-project/ray/pull/25534/files#diff-93d45eb188c58669ee72366fe535cdd468cff865f588b27a4b0f5a027c24ac8d?

This will avoid the circular import, and I just want to make sure we don't lose track of 2 separate DataBatchType

Reply:
Can we actually merge that PR of yours first? Thanks!